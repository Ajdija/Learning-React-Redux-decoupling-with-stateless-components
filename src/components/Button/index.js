import React from 'react';

export const Button = ({ text, buttonClass, clickHandler, spanClass = "title" }) => (
    <button
      className={buttonClass}
      onClick={() => clickHandler ? clickHandler.call(null, text) : null}
    >
        <span className={spanClass}>{text}</span>
    </button>
);

export default Button;
