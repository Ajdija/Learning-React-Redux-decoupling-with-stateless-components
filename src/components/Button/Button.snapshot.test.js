import React from 'react';
import Button from './index';
import renderer from 'react-test-renderer';

describe('button component snapshot test', () => {
  it('takes snapshot of a button component with class and text', () => {
    const json = renderer.create(
      <Button buttonClass="testClass" text="12345" />
    ).toJSON();
    expect(json).toMatchSnapshot();
  });
});
