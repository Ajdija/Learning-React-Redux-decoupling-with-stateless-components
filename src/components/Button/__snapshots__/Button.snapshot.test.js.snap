// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component snapshot test takes snapshot of a button component with class and text 1`] = `
<button
  className="testClass"
  onClick={[Function]}
>
  <span
    className="title"
  >
    12345
  </span>
</button>
`;
